python/nano/src/bigdl/nano/tf/keras/Model.py

```diff
@@ -13,11 +13,71 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
-
-
 import tensorflow as tf
-
+import numpy as np
+from time import time
 
 class Model(tf.keras.Model):
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
+
+    def fit(self,
+            x=None,
+            y=None,
+            batch_size=None,
+            epochs=1,
+            verbose='auto',
+            callbacks=None,
+            validation_split=0.,
+            validation_data=None,
+            shuffle=True,
+            class_weight=None,
+            sample_weight=None,
+            initial_epoch=0,
+            steps_per_epoch=None,
+            validation_steps=None,
+            validation_batch_size=None,
+            validation_freq=1,
+            max_queue_size=10,
+            workers=1,
+            use_multiprocessing=False,
+            perf_tune=False):
+        if perf_tune:
+            if batch_size is None:
+                batch_size = 32
+            # Data pipeline
+            print("Prefetch Datapipeline")
+            assert issubclass(
+                type(x), tf.data.Dataset), "Data object currently only support tf.data.dataset"
+            dataList = [
+                x.take(1).cache().repeat(100),
+                x.take(1).cache().repeat(100).unbatch().batch(int(batch_size/2)),
+                x.take(1).cache().repeat(100).unbatch().batch(int(batch_size*2)),
+            ]
+            
+            for x in dataList:
+                start = time()
+                super(Model, self).fit(
+                x, y, batch_size, epochs, verbose, callbacks, validation_split, validation_data, shuffle, class_weight,
+                    sample_weight, initial_epoch, steps_per_epoch, validation_steps, validation_batch_size, validation_freq,
+                    max_queue_size, workers, use_multiprocessing)
+                end = time()
+                print("Throughput: ", len(x)/(end-start))
```

Comment:
Maybe you do not need to use `len(x)`, I think `len(x)` is just `100 * batch_size`

Reply:
The dataset size become UNKONWN after `unbatch()` called. We cannot get the size of the dataset before the epoch finished: 
![image](https://user-images.githubusercontent.com/42380833/151476114-3aeb59e7-433e-4c82-a4c1-eb394b018f18.png)


After the epoch finished, maybe we can get the size information of the dataset through the progress bar displayed:
![image](https://user-images.githubusercontent.com/42380833/151475624-3014043b-0f4b-43d7-a789-d52fc43b64d3.png)

But I still not sure about the relationship between the dataset size and new batch_size. Should I use hard encoded number when calculating throughput?